packages/vitest/src/node/core.ts

```diff
@@ -462,6 +462,9 @@ export class Vitest {
     }
     const onUnlink = (id: string) => {
       id = slash(id)
+      const needsRerun = this.handleFileChanged(id)
+      if (needsRerun)
+        this.scheduleRerun(id)
```

Comment:
This should be done only if source file was deleted (should be `else if` on line 476)